corehq/util/itertools.py

```diff
@@ -1,38 +0,0 @@
-from __future__ import absolute_import
-from __future__ import unicode_literals
-
-from operator import lt, gt
-
-
-def merge(generator1, generator2, key=None, reverse=False):
-    """Zip two sorted generators into a sorted generator"""
-    # This function can be replaced by heapq.merge in python3
```

Comment:
obviously not important anymore, but for py3 replacements https://github.com/dimagi/commcare-hq/blob/f5b67bff9e9b9f747503b218d1a1fbb9e20c6d78/corehq/sql_db/util.py#L7-L10 is a pretty nice workflow to follow. introduced https://github.com/dimagi/commcare-hq/pull/23340

Reply:
Nice, I like that, particularly the `backports` nomenclature.  I'd be hesitant here since my implementation isn't a perfect replacement (it only supports precisely two generators, whereas stdlib version supports an arbitrary number).

Reply:
yeah. i think the secret is just copy-pasting from python code https://github.com/python/cpython/blob/e42b705188271da108de42b55d9344642170aa2b/Lib/heapq.py#L314 Though for heapq it looks pretty nasty/might not have been possible

Reply:
I considered that actually, but it uses kwargs specified after a `*arg` thingy (I forget what that's called), which isn't possible in python 2, plus there's a lot going on in that file that I didn't want to try to understand.
```python
def merge(*iterables, key=None, reverse=False):
```